### PR TITLE
fix(go): validate trusted proxies before trusting X-Forwarded-For

### LINKED ISSUE #6645: IP spoofing via untrusted X-Forwarded-For header in getRealIP
### Product

Platform/Access controls/Other

### Describe the bug

getRealIP unconditionally trusts the X-Forwarded-For header with no trusted-proxy validation. Any client can set this header to an arbitrary value and spoof their IP address. This affects audit logs and rate-limit checks, meaning attackers can evade rate limiting and forge audit trail entries.

### To Reproduce

1. Make any authenticated API request
2. Set the X-Forwarded-For header to any IP address
3. Observe that the spoofed IP appears in audit logs and is used for rate limiting

### Expected behavior

IP address should only be trusted from X-Forwarded-For when the request comes from a known trusted proxy. Direct client requests should use the connection's remote address.

### Screenshots

_No response_

### Deployment Type

Self-hosted

### Additional context

File: httpinfo.go:31. Found via Faultmark (faultmark.com), an AI code scanner that uses a multi-model debate to verify bugs against actual source code before reporting them.

### PR DESCRIPTION
**Pull request by ssshreyans26**
## Summary

`getRealIP` in `backend-go` unconditionally trusts `X-Forwarded-For` and `X-Real-IP` headers from any client. An attacker connecting directly to the server can spoof their IP, which:
- **Poisons audit logs** with fake source IPs
- **Bypasses rate limiting** by rotating a different fake IP per request

This PR adds `TRUSTED_PROXY_CIDRS` support to the Go backend, matching the Node.js backend's existing implementation (`backend/src/server/plugins/ip.ts`):
- **Strict mode** (env var set): Only trust forwarded headers when `RemoteAddr` is within a configured CIDR range
- **Legacy mode** (env var unset): Preserve current header-trusting behavior for backwards compatibility

Uses the same `TRUSTED_PROXY_CIDRS` env var as the Node.js backend — self-hosted deployments that already have it configured will automatically get the protection in the Go backend too.

## Changes

| File | Change |
|------|--------|
| `config/config.go` | Add `TrustedProxyCIDRs` field, parse comma-separated CIDRs into `[]net.IPNet` |
| `middlewares/httpinfo.go` | `getRealIP` checks `RemoteAddr` against trusted CIDRs before trusting headers |
| `server/server.go` | Pass parsed CIDRs from config into `HTTPInfoMiddleware` |
| `cmd/infisical/main.go` | Pass config to `NewServer` |
| `middlewares/httpinfo_test.go` | 9 test cases: legacy mode, strict mode (trusted/untrusted), multiple CIDRs, edge cases |

## Test plan

- [x] All 9 unit tests pass (`go test ./internal/server/middlewares/`)
- [x] `go vet ./...` clean
- [x] `go build ./...` clean
- [x] `golangci-lint` clean on changed packages (0 issues)
- [ ] CI

Fixes #6645

### CODE REVIEW

File: backend-go/internal/server/middlewares/httpinfo.go [hunk 1]
```diff
@@ -11,40 +11,81 @@ import (
...
+	}
+
+	// Trust forwarded headers (either legacy mode or request is from a trusted proxy).
 	if xff := r.Header.Get("X-Forwarded-For"); xff != "" {
-		// Take the first IP (original client)
 		if before, _, ok := strings.Cut(xff, ","); ok {
 			return strings.TrimSpace(before)
 		}
 		return strings.TrimSpace(xff)
 	}
```
[the remark this answers is not in the document]
Comment (ssshreyans26) [reply backend-go/internal/server/middlewares/httpinfo.go:83]: Addressed in b24137f. Strict mode now walks X-Forwarded-For right-to-left, skipping trusted proxy IPs, and returns the first non-trusted IP. Added test cases for proxy-append attack and multi-hop proxy scenarios.
